Assets/Scripts/WorldEngine/World.cs

```diff
@@ -2606,41 +2606,20 @@ public void PerformTerrainAlterationDrainageRegen(bool doTerrainModification = t
 
                 if (reInsert)
                 {
-                    _drainageHeap.Insert(cell);
+                    _drainageEvalHeap.Insert(cell);
                     continue;
                 }
 
-                if (doTerrainModification)
-                {
-                    cellsToModify.Add(cell);
-                }
+                UpdateWater(cell);
             }
 
             DrainToNeighbors(cell, AddToDrainageRegen);
         }
-
-        if (doTerrainModification)
```

Comment:
Also why the removal of terrain modification?

sidenote: I know you said not to sweat the details, but the drainage feature is complex and I'm just trying to understand.

Reply:
The intention here was to alter a cell's altitude or temperature due to drainage effects. But these alteration should not be done while drainage was still being processed for other cells. Instead, those alterations are now done after all drainage calculations have been completed.